### PR TITLE
chore: review code split docs

### PR DESCRIPTION
**Pull request by zllkjc**
### <samp>🤖 Generated by Copilot at c51f987</samp>

This pull request updates the documentation of the code splitting feature in Modern.js for both English and Chinese readers. It fixes errors, clarifies concepts, and provides more relevant examples using `Suspense`, `React.lazy`, and Loadable. It also adds a changeset file for the `@modern-js/main-doc` package.

### <samp>🤖 Generated by Copilot at c51f987</samp>

*  Add a changeset file to document the changes to the `@modern-js/main-doc` package ([link](https://github.com/web-infra-dev/modern.js/pull/3921/files?diff=unified&w=0#diff-a4304648b3775d6058a2a7990ab52962d6be6d950cd82df474193c2a1c416bf6R1-R6))
*  Update the title and the introduction of the code splitting article in both English and Chinese documentation, fixing a grammatical error and clarifying the meaning of code splitting ([link](https://github.com/web-infra-dev/modern.js/pull/3921/files?diff=unified&w=0#diff-59e6913c1e0835d1fd3d22f8b942087cab77d045468b3d224ee5914c11805996L5-R10), [link](https://github.com/web-infra-dev/modern.js/pull/3921/files?diff=unified&w=0#diff-154bd162b264c60acc26b28583495b12a3ea088bd6f3b39f8223a71cae3b3c4dL10-R10))
*  Simplify the explanation of the dynamic `import()` syntax in both English and Chinese documentation, removing the unnecessary reference to the packaging entry ([link](https://github.com/web-infra-dev/modern.js/pull/3921/files?diff=unified&w=0#diff-59e6913c1e0835d1fd3d22f8b942087cab77d045468b3d224ee5914c11805996L20-R19), [link](https://github.com/web-infra-dev/modern.js/pull/3921/files?diff=unified&w=0#diff-154bd162b264c60acc26b28583495b12a3ea088bd6f3b39f8223a71cae3b3c4dL20-R19))
*  Remove a redundant sentence from both English and Chinese documentation, which repeats the information shown in the code example ([link](https://github.com/web-infra-dev/modern.js/pull/3921/files?diff=unified&w=0#diff-59e6913c1e0835d1fd3d22f8b942087cab77d045468b3d224ee5914c11805996L28-L29), [link](https://github.com/web-infra-dev/modern.js/pull/3921/files?diff=unified&w=0#diff-154bd162b264c60acc26b28583495b12a3ea088bd6f3b39f8223a71cae3b3c4dL28-L29))
*  Update the caution note about the compatibility of `React.lazy` with SSR in both English and Chinese documentation, reflecting the latest status of React 18 Streaming SSR and recommending the Loadable API instead ([link](https://github.com/web-infra-dev/modern.js/pull/3921/files?diff=unified&w=0#diff-59e6913c1e0835d1fd3d22f8b942087cab77d045468b3d224ee5914c11805996L35-R34), [link](https://github.com/web-infra-dev/modern.js/pull/3921/files?diff=unified&w=0#diff-154bd162b264c60acc26b28583495b12a3ea088bd6f3b39f8223a71cae3b3c4dL35-R34))
*  Update the introduction and the code example of the Loadable API in both English and Chinese documentation, specifying the source of the Loadable API and showing how to use it in a Modern.js project ([link](https://github.com/web-infra-dev/modern.js/pull/3921/files?diff=unified&w=0#diff-59e6913c1e0835d1fd3d22f8b942087cab77d045468b3d224ee5914c11805996L61-R61), [link](https://github.com/web-infra-dev/modern.js/pull/3921/files?diff=unified&w=0#diff-154bd162b264c60acc26b28583495b12a3ea088bd6f3b39f8223a71cae3b3c4dL59-R61))
*  Update the explanation and the info note of the Loadable API in both English and Chinese documentation, highlighting the advantages of using the Loadable API in SSR projects and clarifying the limitation of using the Loadable API with `<Suspense>` in SSR and CSR projects ([link](https://github.com/web-infra-dev/modern.js/pull/3921/files?diff=unified&w=0#diff-59e6913c1e0835d1fd3d22f8b942087cab77d045468b3d224ee5914c11805996L75-R81), [link](https://github.com/web-infra-dev/modern.js/pull/3921/files?diff=unified&w=0#diff-154bd162b264c60acc26b28583495b12a3ea088bd6f3b39f8223a71cae3b3c4dL75-R81))

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/document/main-doc/docs/zh/guides/advanced-features/code-split.mdx [hunk 1]
```diff
@@ -17,23 +16,22 @@ sidebar_position: 6
...
-
 ## React.lazy
 
 React 官方提供的组件代码分割的方式。
 
 :::caution
-React 17 及以下版本不支持 SSR，建议 React17 的 SSR 应用使用 loadable。
+`React.lazy` 通常配合 `<Suspense>` 使用，因此仅在 CSR 或 React 18 Streaming SSR 中可用。
 
+对于使用 String SSR 的项目，不支持 `React.lazy`，请使用 Loadable API。
```
Comment (caohuilin) [inline packages/document/main-doc/docs/zh/guides/advanced-features/code-split.mdx:34]: String SSR replace use Traditional SSR？

File: packages/document/main-doc/docs/en/guides/advanced-features/code-split.mdx [hunk 2]
```diff
@@ -2,13 +2,12 @@
 title: Code Split
 sidebar_position: 6
 ---
-# Code Split
+# Code Splitting
 
-Code Split is a common way to optimizing front-end resource loading. This doc will introduce three methods supported by Modern.js:
+Code splitting is a common method to optimize frontend resource loading. This article will introduce the three types of code splitting supported by Modern.js:
```
Comment (yimingjfe) [inline packages/document/main-doc/docs/en/guides/advanced-features/code-split.mdx:7]: Is it better to use "method" than "way"?

Comment (zllkjc) [reply packages/document/main-doc/docs/en/guides/advanced-features/code-split.mdx:7]: if there is a specific way, it is better to use method, chatgpt says

File: packages/document/main-doc/docs/en/guides/advanced-features/code-split.mdx [hunk 3]
```diff
@@ -72,9 +70,12 @@ function MyComponent() {
...
 
-:::info
-SSR is supported out of the box by `loadable`.
+However, it's important to note that any Loadable API in SSR does not support the use of `<Suspense>`.
 
+:::info
+If you want to use `<Suspense>` in CSR projects with React 17 and below, you can substitute `React.lazy` with `loadable.lazy`.
 :::
+
+For detail, see [Loadable API](/apis/app/runtime/utility/loadable).
```
[the remark this answers is not in the document]
Comment (zllkjc) [reply packages/document/main-doc/docs/en/guides/advanced-features/code-split.mdx]: done

**Review by yimingjfe — APPROVED** [review]

**Pull request merged**